lib/index.js

```diff
@@ -0,0 +1,89 @@
+'use strict';
+
+/**
+ * Module Dependencies
+ */
+
+var integration = require('@segment/analytics.js-integration');
+var each = require('@ndhoule/each');
+var foldl = require('@ndhoule/foldl');
+var qs = require('qs');
+
+/**
+ * Expose `DoubleClick Floodlight` integration.
+ */
+
+var Floodlight = module.exports = integration('DoubleClick Floodlight')
+  .option('source', '')
+  .tag('standard', '<img src="http://ad.doubleclick.net/activity;src={{ src }};type={{ type }};cat={{ cat }};ord={{ ord }}?">')
+  .tag('custom', '<img src="http://ad.doubleclick.net/activity;src={{ src }};type={{ type }};cat={{ cat }};ord={{ ord }};{{ customVariables }}?">')
+  .mapping('events');
+
+/**
+ * Initialize.
+ *
+ * https://support.google.com/dcm/partner/answer/4293719?hl=en&ref_topic=4241548
+ * @api public
+ */
+
+Floodlight.prototype.initialize = function() {
+  this.ready();
+};
+
+/**
+ * Track
+ * Fire Floodlight image tags per Segment event
+ *
+ * https://support.google.com/dcm/partner/answer/2823425?hl=en&ref_topic=4241548
+ * https://support.google.com/dcm/partner/answer/2823363
+ * @api public
+ * @param {Track}
+ */
+
+Floodlight.prototype.track = function(track) {
+  // Returns array of objects
+  var mappedEvents = this.events(track.event());
+  var settings = this.options;
+
+  // Must have events mapped and DoubleClick Advertiser ID
+  if (!mappedEvents.length || !settings.source) return;
+  var properties = track.properties();
+  var self = this;
+
+  // Prepare tag params for each mapped Floodlight Activity
+  var tags = foldl(function(conversions, tag) {
+    // Required params. These should already be validated at the app level but just in case
+    if (!tag.event || !tag.cat || !tag.type) return conversions;
+
+    // Find matching properties if any
+    var matchedVariables = {};
+    each(function(variable) {
+      var segmentProp = variable.key;
+      var floodlightProp = variable.value;
+      var segmentPropValue = properties[segmentProp];
+
+      if (segmentPropValue) matchedVariables[floodlightProp] = segmentPropValue;
+    }, tag.customVariable);
+
+    var cacheBuster = Math.random() * 10000000000000000000;
+    var tagParams = {
+      src: settings.source,
+      type: tag.type,
+      cat: tag.cat,
+      ord: cacheBuster,
+      isCustom: tag.customVariable.length, // For conditional tag loading later
+      customVariables: qs.stringify(matchedVariables).replace('&', ';')
```

Comment:
This will only replace the first instance of `&`. You want `.replace(/&/g, ';')` to replace all instances